controllers/reaper/reaper_controller.go

```diff
@@ -484,3 +492,31 @@ func (r *ReaperReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&corev1.Service{}).
 		Complete(r)
 }
+
+func (r *ReaperReconciler) reconcileTrustStoresSecret(ctx context.Context, actualReaper *reaperapi.Reaper, logger logr.Logger) error {
+	sName := reaper.GetTruststoresSecretName(actualReaper.Name)
+	sKey := types.NamespacedName{Namespace: actualReaper.Namespace, Name: sName}
+	s := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      sName,
+			Namespace: actualReaper.Namespace,
+		},
+	}
+	if err := r.Client.Get(ctx, sKey, s); err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info("Creating Reaper's truststore ConfigMap", "ConfigMap", sKey)
+			if err = controllerutil.SetControllerReference(actualReaper, s, r.Scheme); err != nil {
```

Comment:
Issue: I think you need `controllerutil.SetOwnerReference` as well.

Reply:
You do not, SetControllerReference sets OwnerReference. It just sets the "controller" part in the OwnerReference. 

https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/controller/controllerutil/controllerutil.go#L85

Reply:
OK, all good then. We'd set both in some other cases in the codebase, maybe that was wrong.